### PR TITLE
add decorator to submit workflow to wait for pending limit

### PR DESCRIPTION
**Pull request by KPostOffice**
If the user provides a limit for workflows then we run submit workflows through a decorator to throttle workflow creation.

### CODE REVIEW

File: investigator/investigator/common.py [hunk 1]
```diff
@@ -43,7 +43,11 @@ def wait_for_limit(func):
         def inner_func1(*args, **kwargs):
             total_pending = inf
             while total_pending > limit:
-                asyncio.sleep(0.5)
+                #####################################################################################################
+                # to make this awaitable would require a major rework because of the number of sync functions which #
+                # call submit_workflow.                                                                             #
+                #####################################################################################################
+                sleep(0.5)
```
Comment (fridex) [inline investigator/investigator/common.py:50]: Can this be a parameter supplied via env var? Also, 0.5 seems to be too often to me.

**Comment by pacospace** [comment]
@harshad16

**Review by pacospace — APPROVED** [review]
LGTM! :100 thanks

**Pull request merged**